source/mir/math/stat.d

```diff
@@ -420,6 +424,158 @@ unittest
 
+/++
+Centers `slice`, which must be a finite iterable.
+
+By default, `slice` is centered by the mean. A custom function may also be provided
+using `centralTendency`.
+
+Returns:
+    The elements in the slice with the average subtracted from them.
++/
+template center(alias centralTendency = mean!(Summation.appropriate))
+{
+    import mir.ndslice.slice: Slice, SliceKind, sliced, hasAsSlice;
+    /++
+    Params:
+        slice = slice
+    +/
+    auto center(Iterator, size_t N, SliceKind kind)(
+        Slice!(Iterator, N, kind) slice)
+    {
+        import core.lifetime: move;
+        import mir.ndslice.topology: vmap;
+        import mir.ndslice.internal: LeftOp, ImplicitlyUnqual;
+
+        auto m = centralTendency(slice.lightScope);
+        alias T = typeof(m);
+        return slice.move.vmap(LeftOp!("-", ImplicitlyUnqual!T)(m));
+    }
+    
+    /// ditto
+    auto center(T)(T[] array)
+    {
+        return center(array.sliced);
+    }
+
+    /// ditto
+    auto center(T)(T withAsSlice)
+        if (hasAsSlice!T)
+    {
+        return center(withAsSlice.asSlice);
+    }
+}
+
+/// Center vector
+version(mir_test)
+@safe pure nothrow
+unittest
+{
+    import mir.ndslice.slice: sliced;
+    import mir.algorithm.iteration: all;
+    import mir.math.common: approxEqual;
+
+    auto x = [1.0, 2, 3, 4, 5, 6].sliced;
+    assert(x.center.all!approxEqual([-2.5, -1.5, -0.5, 0.5, 1.5, 2.5]));
+    
+    // Can center using different functions
+    assert(x.center!hmean.all!approxEqual([-1.44898, -0.44898, 0.55102, 1.55102, 2.55102, 3.55102]));
+}
+
+/// Center dynamic array
+version(mir_test)
+@safe pure nothrow
+unittest
+{
+    import mir.algorithm.iteration: all;
+    import mir.math.common: approxEqual;
+
+    auto x = [1.0, 2, 3, 4, 5, 6];
+    assert(x.center.all!approxEqual([-2.5, -1.5, -0.5, 0.5, 1.5, 2.5]));
+}
+
+/// Center matrix
+version(mir_test)
+@safe pure
+unittest
+{
+    import mir.ndslice: fuse;
+    import mir.algorithm.iteration: all;
+    import mir.math.common: approxEqual;
+    
+    auto x = [
+        [0.0, 1, 2], 
+        [3.0, 4, 5]
+    ].fuse;
+    
+    auto y = [
+        [-2.5, -1.5, -0.5], 
+        [ 0.5,  1.5,  2.5]
+    ].fuse;
+    
+    assert(x.center.all!approxEqual(y));
+}
+
+/// Column center matrix
+version(mir_test)
+pure @safe
+unittest
+{
+    import mir.algorithm.iteration: all;
+    import mir.math.common: approxEqual;
+    import mir.ndslice: fuse;
+    import mir.ndslice.topology: alongDim, byDim, map;
+
+    auto x = [
+        [20.0, 100.0, 2000.0],
+        [10.0,   5.0,    2.0]
+    ].fuse;
+
+    auto result = [
+        [ 5.0,  47.5,  999],
+        [-5.0, -47.5, -999]
+    ].fuse;
+
+    // Use byDim with map to compute average of row/column.
+    auto xCenterByDim = x.byDim!1.map!center;
+    auto resultByDim = result.byDim!1;
+    size_t i = 0;
+    import std.stdio : writeln;
+    foreach(e; xCenterByDim) {
+        assert(e.all!approxEqual(resultByDim[i]));
+        i++;
+    }
```

Comment:
```
assert(xCenterByDim.all!(all!approxEqual)(resultByDim));
```

or, better for documentation and more universal:

```d
assert(xCenterByDim.equal!(equal!approxEqual)(resultByDim));

```

Reply:
@9il I could not for the life of me figure out how to get that to work in a one-liner. Thanks!